api/broker/broker_controller.go

```diff
@@ -85,94 +97,144 @@ func (ctrl *Controller) addBroker(response http.ResponseWriter, request *http.Re
 	broker.CreatedAt = currentTime
 	broker.UpdatedAt = currentTime
 
-	err = ctrl.BrokerStorage.Create(broker)
+	catalog, err := c.getBrokerCatalog(broker)
+	if err != nil {
+		return err
+	}
+	broker.Catalog = catalog
+
+	err = c.BrokerStorage.Create(broker)
 	err = common.HandleUniqueError(err, "broker")
 	if err != nil {
 		return err
 	}
 
 	broker.Credentials = nil
+	broker.Catalog = nil
 	return rest.SendJSON(response, http.StatusCreated, broker)
 }
 
-func getBrokerID(request *http.Request) string {
-	return mux.Vars(request)[reqBrokerID]
-}
-
-func (ctrl *Controller) getBroker(response http.ResponseWriter, request *http.Request) error {
+func (c *Controller) getBroker(response http.ResponseWriter, request *http.Request) error {
 	brokerID := getBrokerID(request)
 	logrus.Debugf("Getting broker with id %s", brokerID)
 
-	broker, err := ctrl.BrokerStorage.Get(brokerID)
+	broker, err := c.BrokerStorage.Get(brokerID)
 	err = common.HandleNotFoundError(err, "broker", brokerID)
 	if err != nil {
 		return err
 	}
-	broker.Credentials = nil
 
+	broker.Credentials = nil
+	broker.Catalog = nil
 	return rest.SendJSON(response, http.StatusOK, broker)
 }
 
-func (ctrl *Controller) getAllBrokers(response http.ResponseWriter, request *http.Request) error {
+func (c *Controller) getAllBrokers(response http.ResponseWriter, request *http.Request) error {
 	logrus.Debug("Getting all brokers")
-
-	brokers, err := ctrl.BrokerStorage.GetAll()
+	brokers, err := c.BrokerStorage.GetAll()
 	if err != nil {
 		return err
 	}
+	withCatalog := request.FormValue(catalogParam)
+	if strings.ToLower(withCatalog) != "true" {
```

Comment:
yep we would need to rework the storage abit so that you can specify which columns  of the table you need .. This way we won't have to do stuff like credentials = nil and catalog = nil. 
However, I am not sure if we can easily rewrite the db layer this way.

Another option is to have a db layer method getBrokerWithCatalog (but I don't like the approach)   because then we also need GetBrokerWithCatalog, etc etc.. - it becomes too hardcoded.

Basically, I went for the simplest way here - I am open for suggestions, though.